numpy/linalg/tests/test_linalg.py

```diff
@@ -1221,6 +1221,12 @@ class _TestNormBase:
     dt = None
     dec = None
 
+    @staticmethod
+    def check_dtype(a, res):
+        assert_(issubclass(res.dtype.type, np.floating))
+        if issubclass(a.dtype.type, np.inexact):
+            assert_equal(res.dtype, a.real.dtype)
```

Comment:
```suggestion
        assert_equal(res.dtype, a.real.dtype)
```
To clarify: `a.dtype.type` is _always_ a `np.inexact` subclass if the first assertion passes, so might as well directly skip to the second assert.

I suppose that, at this point, you could also consider removing the dedicated assertion method (again), considering it's only a single line of code.
I'll leave that choice up to you though.

Reply:
The existing code `for each_inexact_types in all_types` is misleading.  It runs over integer types, too.

Reply:
Ah, so that's why that `if` statement is there.
Nevermind in that case.

Reply:
A comment to clarify that would be helpful